### ISSUE TITLE
Farm Rush

### ISSUE THREAD

**Issue by Braekpo1nt** [Braekpo1nt/MCTManager#563]
## Description
Game: Farm Rush
Focus: Efficiency of Progression
How it Works: Teams start with an identical plot of land and must harvest ores, animals, and crops. They must turn in items to get points but these items also are a key component of their progression ie wheat for food, breeding, or points/ iron for tools or points
Each item is worth a different amount of points, iron worth less than diamonds, wheat less than meat, that kind of thing. Mobs do not naturally spawn and must be bred. Spawners can also be found underground and farmed for their drops. The underground is also identical on all maps

## Definition of Done
- [x] Farm Rush is implemented
- [x] Features:
  - [x] Items get removed from floor when game is over
  - [x] doesn't sell items that are worth nothing
  - [x] items can be worth money in certain stack counts, but not alone (e.g. 1 point per 5 granite)
  - [x] Items get a Lore that says how many points they are worth when entering the player's inventory or a chest, or perhaps just when an item is spawned in. 
  - [x] `/top`? 
  - [x] Items dropped in the delivery box are sold, give points, and disappear 
  - [x] Creates player areas using WorldEdit API automatically based off of schematic file and description
  - [x] Timer for the entire game
  - [x] Starting out in their barn, 
    - [x] with a glass barrier in the door, 
    - [x] can't leave their area
  - [x] Players quitting
  - [x] Players joining
  - [x] Player damage
- [x] Restrictions:
  - [x] Players can't leave their area
    - [x] teleporting outside or moving outside sends them back to the spawn
  - [x] Players can't break the deposit barrel
  - [x] Players can't do friendly damage
- [x] Config:
  - [x] config is validated
  - [x] Basic common across all configs
  - [x] Durations:
    - [x] Duration of entire game
    - [x] Other standard durations
  - [x] Player Area
    - [x] Schematic file location
    - [x] height limit
    - [x] Dimensions of clone (used to calculate schematic placement)
    - [x] spawn location
    - [x] Starter chest spawn location
    - [x] Deposit Barrel spawn location and orientation
  - [x] Items
    - [x] which items can be dropped off and their value
  - [x] Items in the starter chest
  - [x] Starting player inventory

## Shotgun's description:
Phases
1:
Beginning, each team is teleported to it's own location that it identical to all the others, see FRush world to see what it looks like
2:
A timer aprox 10 minutes begins to count down as the teams each try to gather resources as they gather these resources specific ones can be sold for varying points in the farm house either via barrel or some other method
3:
The game ends after 10 minutes and the points are tallied, team cannot see their score or other teams' scores during this game until this phase, to replace the points segment, total items sold will be displayed until phase 3 where points are tallied and the chat reads off top sellers for each item
Example
Top sellers for ----
1. Red
2. Orange
3. ----

New message with 5 second delay
Top sellers for ----

Repeat

Then scores get added up as the chat reads off until all are done

Map is then reset
The barn interior and structure should not be able to have blocks placed or removed

So basically, they play minecraft for 10 minutes
